### DISCUSSION TITLE
What model removes the vibraslap?

### DISCUSSION THREAD

**Discussion by kbuwel** [nomadkaraoke/python-audio-separator#154] [General]
When the voice and vibra slap are mixed, the vibra slap is not removed, so which model should be used to remove it? I heard that vibra slap is also a musical instrument.

**Comment by beveradb** [comment]
You mean one of these?
<img width="346" alt="image" src="https://github.com/user-attachments/assets/8f3d1a65-f8ad-4f20-8740-e20ea4274e1a">

This is a pretty niche request 😅  - I'd be surprised if anyone had trained a model to specifically separate Vibraslap sounds from an audio file - I had actually never even heard of it before you asked this!

I ended up reading https://en.wikipedia.org/wiki/Vibraslap and watching this to get an idea of the sound: https://www.youtube.com/watch?v=J98Vg-k7_Yg

I guess since it's a percussion instrument it might come out in the "drums" stem from `htdemucs_6s.yaml` or `kuielab_b_drums.onnx` but it's gonna depend on the audio input file you use and how that was mixed, e.g. whether percussion is coming from a different, separate recording and was then mixed in afterwards, or if it was recorded from the same input microphone in the same studio (e.g. with older music).

Unfortunately, in the example clip I tested with, the vibraslap sound seems to get mixed in with the vocal stem - but you can isolate just the vibraslap sound by clipping that section of the vocal stem.
See output files here: [crazy-train-vibraslap-stems.zip](https://github.com/user-attachments/files/18053615/crazy-train-vibraslap-stems.zip)

If you can provide an example input audio file I'll try and give more specific advice for that input; but generally I'd say run it through `htdemucs_6s.yaml` and clip bits of the output stems.
